src/cartesian/ticks/getTicks.ts

```diff
@@ -163,8 +154,10 @@ export function getTicks(props: CartesianAxisProps, fontSize?: string, letterSpa
     return getNumberIntervalTicks(ticks, typeof interval === 'number' && isNumber(interval) ? interval : 0);
   }
 
-  if (interval === 'preserveStartEnd') {
-    return getTicksStart(
+  let candidates: CartesianTickItem[] = [];
+
+  if (interval === 'preserveStart' || interval === 'preserveStartEnd') {
```

Comment:
My personal preference is for the early returns as in before the changes, but since you moved the _isShow_ filtering here it makes sense to try and reduce code duplication.

However it could also make sense that to not stop here but create and use an additional layer like
```typescript
function getShownTicks(...) {
  return getTicks(...).filter(entry => entry.isShow);
}
```

But then you have to go and change every usage of getTicks to keep existing behaviour (and it probably does not work too well with the following PR?).

Alternatively, if we would want to go one step further in reducing code duplication we could introduce a holder for these like
```typescript
const tickGeneratorForInterval: Record<
  CartesianAxisProps['interval'],
  (args: Omit<CartesianAxisProps, 'tickMargin'>) => CartesianTickItem[]
> = { 
  'preserveEnd': getTicksEnd, 
  'preserveStart': getTicksStart, 
  'preserveStartEnd': (args) => getTicksStart(args, true),
}
```
and then just
```typescript
  const candidates = tickGeneratorForInterval[interval]({ ... });
  return candidates.filter(entry => entry.isShow);
```
to ~~get rid of~~ hide the branching altogether

Reply:
I like the ideas, and will most probably come back to them - but only when following up with this file after landing the feature enhancement. For now I want to only refactor as much as I need to, in order to nicely implement the equidistant ticks.

Reply:
Looking at the following PR this also has the benefit that adding the new option is just
```typescript
  'equidistantPreserveStart': (args) => getEveryNthTick(getTicksStart(args)),
```
instead of adding back the duplicated code that this PR removed. With that context I would argue even more so that the conditions should still be separate and not share the branch for 'preserveStart' and 'preserveStartEnd'.